jaeger-core/src/main/java/io/jaegertracing/spi/BaggageRestrictionManager.java

```diff
@@ -12,7 +12,9 @@
  * the License.
  */
 
-package io.jaegertracing.baggage;
+package io.jaegertracing.spi;
+
+import io.jaegertracing.internal.baggage.Restriction;
```

Comment:
What should we do in these cases - the Restriction class is part of the SPI - we shouldn't really be returning or passing instances of internal classes in a public API?

Reply:
Last Friday we talked briefly about this: providers (those who will consume the SPI) are expected to depend on internal classes, like `JaegerSpan` (we discussed it when talking about the `Reporter`s and `Sender`s).

End consumes of our code will only ever be required to deal with the `Configuration` object and OpenTracing API, at least for this initial version. If we later decide that some class should really be part of the public API, we'll white list that specific one.

Reply:
Yeah agree that was what we discussed - but on reflection I'm thinking that if a class is part of the "public" API/SPI, then it should also be public - as end users may contribute SPI implementations.
I'll have a look through the current SPI interfaces to see how many current internal classes are affected - if not many, it may be worth moving them to be public again.

Reply:
> as end users may contribute SPI implementations.

Then they are wearing the "provider" hat on this code/module. Their "consumer" code shouldn't make use of the `Restriction` class, only their provider implementation. And provider implementations are expected to deal with internal classes.

It's still not my first choice, as I'd rather have clear compatibility expectations not only for end consumers (public API) but also for providers: as a provider, can I rely on the `getTracer()` method from `JaegerSpan`? 

*But*, the current arrangement is already better than the status quo :)

Reply:
Not sure I agree with providers having to deal with internal classes. If the provider is defined independent of the jaeger code base, it should expect a stable SPI, which is not guaranteed in relation to the internal classes.

+1 I would prefer `JaegerSpan` was a public interface, with internal `JaegerSpanImpl` - even if for now `JaegerSpan` had no other methods other than derived from `Span`.

Reply:
> it should expect a stable SPI, which is not guaranteed in relation to the internal classes.

Agreed. 

> +1 I would prefer JaegerSpan was a public interface, with internal JaegerSpanImpl - even if for now JaegerSpan had no other methods other than derived from Span.

The other suggestion made during the call was to indeed have this split, but with empty interfaces, like `public interface JaegerSpan extends Span {}`.